### ISSUE TITLE
[Bug] In-game LUA errors upon logging in

### ISSUE THREAD

**Issue by towneh** [Alumian/BCEPGP#1]
Using the latest clone of the master branch

```
Message: [string "BCEPGP_frame:OnEvent"]:1: attempt to call global 'BCEPGP_OnEvent' (a nil value)
Time: 02/16/19 16:49:53
Count: 1
Stack: [string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>
[C]: ?
Interface\AddOns\!Compatibility\errorHandler.lua:14: in function <Interface\AddOns\!Compatibility\errorHandler.lua:11>
[C]: ?
[C]: in function `pairs'
Interface\AddOns\BCEPGP\Button.lua:226: in function `initFunction'
Interface\FrameXML\UIDropDownMenu.lua:54: in function `UIDropDownMenu_Initialize'
[string "*:OnLoad"]:1: in function <[string "*:OnLoad"]:1>
```
 The !Compatibility addon is from the latest clone of https://github.com/ElvUI-TBC/ElvUI

---

```
Message: Interface\AddOns\BCEPGP\Button.lua:226: bad argument #1 to 'pairs' (table expected, got nil)
Time: 02/16/19 16:49:53
Count: 1
Stack: [C]: in function `pairs'
Interface\AddOns\BCEPGP\Button.lua:226: in function `initFunction'
Interface\FrameXML\UIDropDownMenu.lua:54: in function `UIDropDownMenu_Initialize'
[string "*:OnLoad"]:1: in function <[string "*:OnLoad"]:1>
```

**Comment by Alumian** [comment]
Hi Towneh,

Thanks for the report. Could you please advise what you were doing when the error occured?

Thanks,
Alumian

**Comment by Alumian** [comment]
Hi Towneh,

Does this error occur every time upon logging in for you?

Thanks,
Alumian

**Comment by towneh** [comment]
Yep, happens every time I log in.

**Comment by towneh** [comment]
```

CHANNEL = "GUILD"
MOD = 1
AUTOEP = {
	["The Curator"] = true,
	["Brutallus"] = true,
	["Gurtogg Bloodboil"] = true,
	["Attumen the Huntsman"] = true,
	["Illidan Stormrage"] = true,
	["Moroes"] = true,
	["Rage Winterchill"] = true,
	["Terestian Illhoof"] = true,
	["Prince Malchezaar"] = true,
	["Morogrim Tidewalker"] = true,
	["Kil'jaeden"] = true,
	["Anetheron"] = true,
	["Maiden of Virtue"] = true,
	["Netherspite"] = true,
	["Shade of Aran"] = true,
	["Kalecgos"] = true,
	["Lady Vashj"] = true,
	["Archimonde"] = true,
	["Leotheras the Blind"] = true,
	["Teron Gorefiend"] = true,
	["The Lurker Below"] = true,
	["Nightbane"] = true,
	["Illidari Council"] = true,
	["Felmyst"] = true,
	["Azgalor"] = true,
	["The Eredar Twins"] = true,
	["Kaz'rogal"] = true,
	["Fathom-Lord Karathress"] = true,
	["Reliquary of Souls"] = true,
	["High King Maulgar"] = true,
	["High Warlord Naj'entus"] = true,
	["The Opera Event"] = true,
	["M'uru"] = true,
	["Shade of Akama"] = true,
	["The Chess Event"] = true,
	["Hydross the Unstable"] = true,
	["Void Reaver"] = true,
	["Al'ar"] = true,
	["Supremus"] = true,
	["Kael'thas Sunstrider"] = true,
	["High Astromancer Solarian"] = true,
	["Mother Shahraz"] = true,
	["Tenris Mirkblood"] = true,
	["Gruul the Dragonkiller"] = true,
	["Magtheridon"] = true,
}
EPVALS = {
	["The Curator"] = 3,
	["Brutallus"] = 13,
	["Gurtogg Bloodboil"] = 11,
	["Attumen the Huntsman"] = 3,
	["Illidan Stormrage"] = 15,
	["Moroes"] = 3,
	["Rage Winterchill"] = 8,
	["Terestian Illhoof"] = 4,
	["Prince Malchezaar"] = 5,
	["Morogrim Tidewalker"] = 6,
	["Kil'jaeden"] = 18,
	["Anetheron"] = 8,
	["Maiden of Virtue"] = 3,
	["Netherspite"] = 4,
	["Shade of Aran"] = 4,
	["Kalecgos"] = 12,
	["Lady Vashj"] = 8,
	["Archimonde"] = 11,
	["Leotheras the Blind"] = 4,
	["Teron Gorefiend"] = 10,
	["The Lurker Below"] = 4,
	["Nightbane"] = 5,
	["Illidari Council"] = 12,
	["Felmyst"] = 13,
	["Azgalor"] = 9,
	["The Eredar Twins"] = 14,
	["Kaz'rogal"] = 9,
	["Fathom-Lord Karathress"] = 6,
	["Reliquary of Souls"] = 11,
	["High King Maulgar"] = 4,
	["High Warlord Naj'entus"] = 10,
	["The Opera Event"] = 3,
	["M'uru"] = 14,
	["Shade of Akama"] = 10,
	["The Chess Event"] = 4,
	["Hydross the Unstable"] = 4,
	["Void Reaver"] = 7,
	["Al'ar"] = 7,
	["Supremus"] = 10,
	["Kael'thas Sunstrider"] = 10,
	["High Astromancer Solarian"] = 7,
	["Mother Shahraz"] = 12,
	["Tenris Mirkblood"] = 3,
	["Gruul the Dragonkiller"] = 5,
	["Magtheridon"] = 5,
}
BASEGP = 1
STANDBYEP = false
STANDBYOFFLINE = false
STANDBYPERCENT = 0
STANDBYRANKS = {
	{
		"Scarab Lord", -- [1]
		false, -- [2]
	}, -- [1]
	{
		"Officer", -- [1]
		false, -- [2]
	}, -- [2]
	{
		"Officer alt", -- [1]
		false, -- [2]
	}, -- [3]
	{
		"Top Lad", -- [1]
		false, -- [2]
	}, -- [4]
	{
		"Academic", -- [1]
		false, -- [2]
	}, -- [5]
	{
		"Top Lad Intern", -- [1]
		false, -- [2]
	}, -- [6]
	{
		"Alt", -- [1]
		false, -- [2]
	}, -- [7]
	{
		"Soggy BottomBoy", -- [1]
		false, -- [2]
	}, -- [8]
	{
		"", -- [1]
		false, -- [2]
	}, -- [9]
	{
		"", -- [1]
		false, -- [2]
	}, -- [10]
}
SLOTWEIGHTS = {
	["2HWEAPON"] = 2,
	["NECK"] = 0.5,
	["HEAD"] = 1,
	["WEAPON"] = 1.5,
	["WRIST"] = 0.5,
	["WEAPONMAINHAND"] = 1.5,
	["SHIELD"] = 0.5,
	["HOLDABLE"] = 0.5,
	["RANGED"] = 0.5,
	["THROWN"] = 0.5,
	["FEET"] = 0.75,
	["RELIC"] = 0.5,
	["RANGEDRIGHT"] = 0.5,
	["WEAPONOFFHAND"] = 0.5,
	["TRINKET"] = 0.75,
	["FINGER"] = 0.5,
	["CLOAK"] = 0.5,
	["CHEST"] = 1,
	["ROBE"] = 1,
	["HAND"] = 0.75,
	["WAIST"] = 0.75,
	["EXCEPTION"] = 1,
	["LEGS"] = 1,
	["SHOULDER"] = 0.75,
}
RECORDS = {
}
OVERRIDE_INDEX = {
}
TRAFFIC = {
	{
		"RRaid", -- [1]
		"Add Raid EP +4 - High King Maulgar", -- [2]
		"", -- [3]
		"", -- [4]
		"", -- [5]
		"", -- [6]
	}, -- [1]
	{
		"RRaid", -- [1]
		"Add Raid EP +3 - Attumen the Huntsman", -- [2]
		"", -- [3]
		"", -- [4]
		"", -- [5]
		"", -- [6]
	}, -- [2]
	{
		"RRaid", -- [1]
		"Add Raid EP +3 - Moroes", -- [2]
		"", -- [3]
		"", -- [4]
		"", -- [5]
		"", -- [6]
	}, -- [3]
	{
		"RRaid", -- [1]
		"Add Raid EP +3 - Maiden of Virtue", -- [2]
		"", -- [3]
		"", -- [4]
		"", -- [5]
		"", -- [6]
	}, -- [4]
	{
		"RRaid", -- [1]
		"Add Raid EP +3 - The Opera Event", -- [2]
		"", -- [3]
		"", -- [4]
		"", -- [5]
		"", -- [6]
	}, -- [5]
}
BCEPGP_standby_accept_whispers = false
BCEPGP_standby_byrank = true
BCEPGP_standby_manual = false
BCEPGP_standby_whisper_msg = "!standby"

```

**Comment by Alumian** [comment]
Thanks for uploading that. Annoyingly it looks like my initial assumption wasn't the problem. Just calling it a night now but I'll push a potential fix tomorrow (Australia time). I'll post back when I've pushed the update. In the mean time, can you please try disabling all add-ons except BCEPGP and ElvUi and let me know if it still happens? It's really weird because I'm running ElvUi as well but I can't replicate the issue.

**Comment by towneh** [comment]
Yep it's exactly the same error message when disabling all addons except leaving the following enabled:

!Compatiiblity
!DebugTools
BCEPGP
ElvUI
ElvUI_Config

**Comment by Alumian** [comment]
Okay, could you please try renaming your BCEPGP.lua saved variables file, relaunch the game and tell me if the issue persists?

**Comment by towneh** [comment]
Deleted the lua file completed and logged in:-

Message: [string "BCEPGP_frame:OnEvent"]:1: attempt to call global 'BCEPGP_OnEvent' (a nil value)
Time: 02/18/19 18:09:12
Count: 1
Stack: [string "*:OnEvent"]:1: in function <[string "*:OnEvent"]:1>
[C]: ?
Interface\AddOns\!Compatibility\errorHandler.lua:14: in function <Interface\AddOns\!Compatibility\errorHandler.lua:11>
[C]: ?
[C]: in function `pairs'
Interface\AddOns\BCEPGP\Button.lua:226: in function `initFunction'
Interface\FrameXML\UIDropDownMenu.lua:54: in function `UIDropDownMenu_Initialize'
[string "*:OnLoad"]:1: in function <[string "*:OnLoad"]:1>

Message: Interface\AddOns\BCEPGP\Button.lua:226: bad argument #1 to 'pairs' (table expected, got nil)
Time: 02/18/19 18:09:12
Count: 1
Stack: [C]: in function `pairs'
Interface\AddOns\BCEPGP\Button.lua:226: in function `initFunction'
Interface\FrameXML\UIDropDownMenu.lua:54: in function `UIDropDownMenu_Initialize'
[string "*:OnLoad"]:1: in function <[string "*:OnLoad"]:1>

**Comment by towneh** [comment]
Nope I downloaded BCEPGP-master.zip on the 12th feb from this site, I haven't amended any code,

**Comment by Alumian** [comment]
Okay, I pushed an update last night. It didn't specifically contain a fix for this but it did contain other fixes. One change that i made in the update should give me a bit of insight into this issue. Could you download it for me and see if the error still occurs please? Also please post the error message again if it does occur, I need to see if it's exactly the same or not

**Comment by towneh** [comment]
Installed new version, deleted the LUA cache in WTF and I now only receive:

Message: Interface\AddOns\BCEPGP\Utility.lua:734: attempt to index global 'BCEPGP_button_options' (a nil value)
Time: 02/18/19 22:01:10
Count: 1
Stack: Interface\AddOns\BCEPGP\Utility.lua:734: in main chunk

**Comment by Alumian** [comment]
Okay I've just had another confirmed case of this within my guild, I'll try to replicate the issue when I get home today.

**Comment by Alumian** [comment]
Alright, I just pushed an update. Can you please download it and see if the issue is resolved? God I am not proud of that update lmao.

**Comment by towneh** [comment]
Solved! no more errors now :)

**Comment by Alumian** [comment]
Fantastic, thanks for raising this ticket. I'm going to go ahead and close it a a guild member has also confirmed this resolution.

**Comment by Alumian** [comment]
Resolved in 0efcb90910144f48fb596adf918dacba0e9b6bd3.
